src/chainstate/stacks/db/accounts.rs

```diff
@@ -214,14 +214,19 @@ impl StacksChainState {
     ) {
         clarity_tx
             .with_clarity_db(|ref mut db| {
-                let balance = STXBalance::initial(amount);
+                let mut balance = STXBalance::initial(amount);
                 let total_balance = balance.get_total_balance();
 
                 let mut snapshot = db.get_stx_balance_snapshot_genesis(principal);
+                let existing_balance = snapshot.balance().amount_unlocked;
+                if existing_balance > 0 {
+                    balance.amount_locked = balance
```

Comment:
This is incorrect.  `amount_locked` refers to tokens locked through PoX, and has nothing to do with tokens unlocked through the `lockup` contract.

Reply:
This is a typo, should be incrementing `amount_unlocked`, just fixed it. The change to this fn should just result in an increment of balance rather than replacement of balance (liquid, not locked). Does it look correct to you now?